### PR TITLE
fixed-bytes: add `serde`, document feature flags

### PR DESCRIPTION
**Pull request by hinto-janai**
### What
Adds a `serde` feature flag (enabled by default) to `cuprate-fixed-bytes` and documents it.

### Why
For https://github.com/Cuprate/cuprate/pull/220#discussion_r1672904392.

### CODE REVIEW

**Comment by hinto-janai** [comment]
@Boog900 For another PR: it seems backwards to have the `{ByteArray,ByteArrayVec}` epee implementations in `epee-encoding` itself, they should be in `fixed-bytes`; `epee-encoding` could expose some things so it's less cumbersome to do this.

File: net/fixed-bytes/src/lib.rs [hunk 1]
```diff
@@ -43,6 +49,7 @@ impl Debug for FixedByteError {
 /// Internally this is just a wrapper around [`Bytes`], with the constructors checking that the length is equal to `N`.
 /// This implements [`Deref`] with the target being `[u8; N]`.
 #[derive(Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
```
Comment (Boog900) [inline net/fixed-bytes/src/lib.rs:52]: the `Serialize` is probably ok but the `Deserialize` has to make sure the `ByteArray` is a multiple of `N`

File: net/fixed-bytes/src/lib.rs [hunk 2]
```diff
@@ -88,6 +95,7 @@ impl<const N: usize> TryFrom<Vec<u8>> for ByteArray<N> {
 }
 
 #[derive(Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
```
Comment (Boog900) [inline net/fixed-bytes/src/lib.rs:98]: same here

**Comment by Boog900** [comment]
@hinto-janai yeah true, it was from when the trait was sealed

**Review by Boog900 — APPROVED** [review]

**Pull request merged**